### DISCUSSION TITLE
Crud-service views: can I search by encrypted fields?

### DISCUSSION THREAD

**Discussion by edocrippaofficial** [mia-platform/community#269] [Q&A] [answered]
### Which section?

Console

### Which is the version?

5.4.2

### Description

Hi, I have a collection with an encrypted field `name` with the 'searchable' property active, and it works great.

I can search by `name` and it returns me only the documents in the collection with the correct value for `name`.

Then I have a view originating from this collection, including the encrypted field. 

I can see the decrypted field in the results but I cannot search by `name` like in the original collection: it returns me an empty array.

Am I missing something? Or in the views there is no way to search by encrypted fields?

**Comment by danibix95** [comment, marked as answer]
Hi @EdoCrippaOfficial,

currently we haven't verified yet the support of search over encrypted fields in case of views. However, we are planning to verify whether this operation is actually feasible or not.
In the meanwhile I would recommend you to upgrade CRUD Service to the latest version.

**Reply by edocrippaofficial** [reply]
Thanks @danibix95, this is not super urgent but we will need that functionality. 
In the meantime we could use a BFF to proxy the request to the original collection and apply the logic of the aggregation pipeline, so that the encrypted field can be searchable, but a view would be perfect in that spot.

Thanks,
Edo

**Reply by danibix95** [reply]
Hi @EdoCrippaOfficial , unfortunately querying encrypted fields via a MongoDB view is not supported at MongoDB level, as stated in their [documentation](https://www.mongodb.com/docs/manual/core/csfle/reference/limitations/#views).

Therefore, neither CRUD Service can provide support to create views on encrypted fields.
